.ci/test_r_package_windows.ps1

```diff
@@ -132,18 +141,30 @@ Write-Output "Building R package"
 
 # R CMD check is not used for MSVC builds
 if ($env:COMPILER -ne "MSVC") {
-  Run-R-Code-Redirect-Stderr "commandArgs <- function(...){'--skip-install'}; source('build_r.R')"; Check-Output $?
 
-  $PKG_FILE_NAME = Get-Item *.tar.gz
-  $PKG_FILE_NAME = $PKG_FILE_NAME -replace '[\\]', '/'
+  $PKG_FILE_NAME = "lightgbm_$env:LGB_VER.tar.gz"
   $LOG_FILE_NAME = "lightgbm.Rcheck/00check.log"
 
+  if ($env:R_BUILD_TYPE -eq "cmake") {
+    Run-R-Code-Redirect-Stderr "commandArgs <- function(...){'--skip-install'}; source('build_r.R')"; Check-Output $?
+  } elseif ($env:R_BUILD_TYPE -eq "cran") {
+    sh build-cran-package.sh ; Check-Output $?
+    # Test CRAN source .tar.gz in a directory that is not this repo or below it.
+    # When people install.packages('lightgbm'), they won't have the LightGBM
+    # git repo around. This is to protect against the use of relative paths
+    # like ../../CMakeLists.txt that would only work if you are in the repoo
+    $R_CMD_CHECK_DIR="tmp-r-cmd-check"
+    New-Item -Path "C:\" -Name $R_CMD_CHECK_DIR -ItemType "directory" > $null
```

Comment:
We already have to install Rtools at `C:/Rtools`. See this link from line 54 of this file: https://stackoverflow.com/questions/39090983/rcpp-rtools-installed-but-error-message-g-not-found/46619260#46619260

And since that is working, it must be that `C:` is writable in this CI.

Reply:
Ah, I see! It is really not good 😬 . But OK, if it is needed for rtools.

Reply:
yeah I agree with you. There are a lot of, let's say, "interesting" choices in the R-for-Windows toolchain haha